src/enrichment_tables/file.rs

```diff
@@ -169,7 +253,63 @@ impl File {
 
         index.shrink_to_fit();
 
-        index
+        Ok(index)
+    }
+
+    /// Sequentially searches through the iterator for the given condition
+    fn sequential<'a, I>(
+        &'a self,
+        data: I,
+        condition: &'a [Condition<'a>],
+    ) -> impl Iterator<Item = BTreeMap<String, Value>> + 'a
+    where
+        I: Iterator<Item = &'a Vec<Value>> + 'a,
+    {
+        data.filter_map(move |row| {
+            if self.row_equals(condition, &*row) {
+                Some(self.add_columns(&*row))
+            } else {
+                None
+            }
+        })
+    }
+}
+
+/// Adds the bytes from the given value to the hash.
+/// Each field is terminated by a `0` value to separate the fields
+fn hash_value(hasher: &mut seahash::SeaHasher, value: &Value) -> Result<(), String> {
+    match value {
+        Value::Bytes(bytes) => {
+            hasher.write(
+                std::str::from_utf8(bytes)
+                    .map_err(|_| "column contains invalid utf".to_string())?
+                    .to_lowercase()
+                    .as_bytes(),
+            );
+        }
```

Comment:
Looking at the case below, isn't the `hasher` already able to `write` `bytes::Bytes`? Is it necessary to validate for UTF-8 conformance here?

Reply:
It is needed in order to convert to lowercase that is there. I have another PR coming up that makes the case insensitive search optional, for that case I wouldn't need this.

Reply:
Oh, right! Missed that one